web/crux-ui/src/hooks/use-filters.ts

```diff
@@ -67,3 +67,24 @@ export const textFilterFor =
       return properties.filter(it => !!it && it.toLowerCase().includes(text)).length > 0
     })
   }
+
+export type DateRangeFilter = {
+  dateRange: [Date, Date]
+}
+
+export const dateRangeFilterFor =
+  <Item>(propertiesOf: PropertyValuesOf<Item>): FilterFunction<Item, DateRangeFilter> =>
+  (items: Item[], filter: DateRangeFilter) => {
+    if (filter.dateRange[0] === null && filter.dateRange[1] === null) {
+      return items
+    }
+
+    const from = filter.dateRange[0] === null ? new Date(-8640000000000000) : filter.dateRange[0] // Min Date
+    const to = filter.dateRange[1] === null ? new Date(8640000000000000) : filter.dateRange[1] // Max Date
+    if (filter.dateRange[1] !== null) to.setMilliseconds(to.getMilliseconds() + 86399999) // Set to end of the day
```

Comment:
This could be misleading. It only works if the date range contains *only* dates and no time (00:00:00). I think the caller should handle this case not the filter.